spec/lib/workflow/stores/mongo-spec.js

```diff
@@ -643,4 +644,35 @@ describe('Task Graph mongo store interface', function () {
             );
         });
     });
+
+    it('deleteTaskByName', function() {
+        var task = { injectableName: 'testname' };
+        task.toJSON = function() { return this; };
+        waterline.taskdefinitions.destroy.resolves(task);
+        return mongo.deleteTaskByName('testname')
+        .then(function() {
+            expect(waterline.taskdefinitions.destroy).to.have.been.calledOnce;
+            expect(waterline.taskdefinitions.destroy).to.have.been.calledWith(
+                { injectableName: 'testname' }
+            );
+        });
+    });
+
+    it('deleteTaskByName should fail on an invalid query(injectable name)', function() {
```

Comment:
@srinia6 didn't we want to get rid of this negative test since it doesn't return the error properly?